Sources/NIOPerformanceTester/ByteBufferBenchmark.swift

```diff
@@ -0,0 +1,47 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import NIO
+
+final class ByteBufferBenchmark: Benchmark {
+    private let iterations : Int
+    private let bufferSize : Int
+    private var circularBuffer: CircularBuffer<UInt8>
+    private var buffer: ByteBuffer
+
+    init(iterations: Int, bufferSize: Int) {
+        self.iterations = iterations
+        self.bufferSize = bufferSize
+        self.circularBuffer = CircularBuffer<UInt8>(initialCapacity: self.bufferSize)
+        self.buffer = ByteBufferAllocator().buffer(capacity: self.bufferSize)
```

Comment:
Nit: we should probably reserve `bufferSize * iterations` space, given that we call `writeBytes` in the loop. Alternatively, the loop could call `clear()` on each iteration.

Reply:
I added clear()